staging/src/k8s.io/apiserver/pkg/server/options/serving.go

```diff
@@ -165,9 +165,16 @@ func (s *SecureServingOptions) applyServingInfoTo(c *server.Config) error {
 		return nil
 	}
 
+	ipMax := c.MaxRequestsInFlight / 2
+	if ipMax > c.MaxMutatingRequestsInFlight/2 {
+		ipMax = c.MaxMutatingRequestsInFlight / 2
+	}
+	max := c.MaxRequestsInFlight + c.MaxMutatingRequestsInFlight
```

Comment:
again, this max doesn't take into account long-running requests (which don't currently fall into either the maxrequests or maxmutatingrequests buckets)